docker/docker-prepare.sh

```diff
@@ -89,6 +89,46 @@ superuser() {
 	fi
 }
 
+custom_container_init() {
+	# Mostly borrowed from the LinuxServer.io base image
+	# https://github.com/linuxserver/docker-baseimage-ubuntu/tree/bionic/root/etc/cont-init.d
+	readonly custom_script_dir="/custom-cont-init.d"
```

Comment:
A readonly variable is still global per default, so there's risk of failing to declare the variable if it's already present.

My advice, make a local readonly variable:
local -r custom_script_dir="/custom-cont-init.d"

Reply:
Hey it’s schnuffle! 👋

Reply:
Nice, this could probably be added a few more places in the various scripts too

Reply:
Yeah, been back for a while but as you guys are doing really well without me and I'm loaded with many other stuff, I kept a low profile here :).

BAck to the topic, of course another option is to have a global variable, but then it should be declared at the top.
I saw that the use of global/local variable doesn't seem to be stringent in the rest of the script as well, which should be corrected to avoid side effects

Reply:
Alright, I've added `local -r` to things around the scripts.  I can't test to resulting image atm, but will try to this evening.